### ISSUE TITLE
no-unsafe-assignment: array concat false positive

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#30]
```typescript
  const arr: ReadonlyArray<never> = [];
  const foo = arr.concat(arr); // flagged; shouldn't be
```

**Comment by danielnixon** [comment]
All `...rest[]` params are the same because they're considered a (mutable) array.
